src/main/java/executor/service/service/impl/JSONFileProxyProvider.java

```diff
@@ -0,0 +1,38 @@
+package executor.service.service.impl;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import executor.service.model.ProxyConfigHolder;
+import executor.service.service.ProxyProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+public class JSONFileProxyProvider implements ProxyProvider {
+
+    private static final Logger log = LoggerFactory.getLogger(ProxyProvider.class);
+    private static final String FILE_NAME = "/proxy.json";
+
+    @Override
+    public List<ProxyConfigHolder> readProxy() {
```

Comment:
1. Reading ProxyNetworkConfig, ProxyCredentials from different files: proxy-network.json, proxy-credential.json:
![image](https://github.com/GFLCourses5/worker/assets/140727960/3b194a14-f16c-477f-957f-fb129403b613)

2. As a result, it is necessary to ensure that the read ProxyNetworkConfig and ProxyCredentials are correctly combined into the ProxyConfigHolder (the problem may be a different number of elements in the proxy-network.json, proxy-credential.json files).

3. It is also necessary to simulate the situation of random data arrival for implementation in ParralelFlowExecutorService.

In my opinion, pull-request #43 is more suitable for these tasks.

Reply:
THanks again, I have access to Trello. 
1) There you return Flux
2) Ok, if you want 2 files you can do it. But please provide the following workflow:
a) read proxy network -> b) read proxy credentials -> c) combine 2 entities to *ProxyConfigHolder*
but exactly not like you did! There you just throw away some information! Thats not the way it works